### PR TITLE
sending message through query after receiving package from gitlab

### PR DESCRIPTION
**Pull request by sardhara-preet**

### CODE REVIEW

File: api/gitlab/receive-message.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    [GITLAB_ENTITY.PUSH]: 'push',
+    [GITLAB_ENTITY.TAG_PUSH]: 'tag_push',
+    [GITLAB_ENTITY.ISSUE]: 'issue',
+    [GITLAB_ENTITY.MERGE_REQUEST]: 'merge_request',
+    [GITLAB_ENTITY.PIPELINE]: 'pipeline',
+    [GITLAB_ENTITY.JOB]: 'job',
+}
+
+router.post('/', function(req,res) {
+    const messageString = messageCreater[SLACK_ENTITY_TO_BOT_API[req.body.object_kind]](req.body);
```
Comment (dvijeshpatel) [inline api/gitlab/receive-message.js]: does it comes in object_kind?

File: api/gitlab/message-creator.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const messageCreater = {
+
+    push: function(req) {
```
Comment (dvijeshpatel) [inline api/gitlab/message-creator.js]: need to check message this.. message structure

File: api/gitlab/receive-message.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+const router = express.Router();
+router.use(bodyParser.json());
+router.use(bodyParser.urlencoded({ extended: false }));
+
+const SLACK_ENTITY_TO_BOT_API = {
+    [GITLAB_ENTITY.PUSH]: 'push',
+    [GITLAB_ENTITY.TAG_PUSH]: 'tag_push',
+    [GITLAB_ENTITY.ISSUE]: 'issue',
+    [GITLAB_ENTITY.MERGE_REQUEST]: 'merge_request',
```
Comment (dvijeshpatel) [inline api/gitlab/receive-message.js]: all cases are not covered?

File: api/gitlab/message-sender.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+const request = require("request");
+
+module.exports = function(data) {
+
+    request
+        .get('http://preetkumar.localhost.run/slack/sendmessage').form(data)
+        .on('response', function(response) {
```
Comment (dvijeshpatel) [inline api/gitlab/message-sender.js]: what are you doing here?

File: api/gitlab/message-sender.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+const request = require("request");
+
+module.exports = function(data) {
+
+    request
```
Comment (dvijeshpatel) [inline api/gitlab/message-sender.js]: is it message sender or receiver?

Comment (sardhara-preet) [reply api/gitlab/message-sender.js]: sender to our slack API

File: api/gitlab/message-sender.js (hunk 4 above)
Comment (sardhara-preet) [reply api/gitlab/message-sender.js]: sending the get request to our slack API

File: api/gitlab/receive-message.js (hunk 3 above)
Comment (sardhara-preet) [reply api/gitlab/receive-message.js]: I don't know the exact structure of the message which should be formed so I just made an sample.

File: api/gitlab/receive-message.js (hunk 1 above)
Comment (sardhara-preet) [reply api/gitlab/receive-message.js]: It comes as a string

File: api/gitlab/receive-message.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+router.use(bodyParser.json());
+router.use(bodyParser.urlencoded({ extended: false }));
+
+const SLACK_ENTITY_TO_BOT_API = {
+    [GITLAB_ENTITY.NOTE.MERGEREQUEST]: 'note_MergeRequest',
+}
+
+router.post('/', function(req,res) {
+    if(req.body.object_kind === 'note') {
+        const data = messageCreater[SLACK_ENTITY_TO_BOT_API[req.body.object_attributes.noteable_type]](req.body);
```
Comment (dvijeshpatel) [inline api/gitlab/receive-message.js]: is this bot API?

File: api/gitlab/message-creator.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+const findName = require('./name-fetcher');
+
+const messageCreater = {
+
+    note_MergeRequest: function(req) {
+        const obj = {};
+        obj.type = 'user';
```
Comment (dvijeshpatel) [inline api/gitlab/message-creator.js]: how you decided type should be user?

File: api/gitlab/message-sender.js [hunk 8]
```diff
@@ -0,0 +1,10 @@
...
+        .get('http://preetkumar.localhost.run/slack/sendmessage').form(data)
```
Comment (dvijeshpatel) [inline api/gitlab/message-sender.js]: use Process variable

Comment (dvijeshpatel) [reply api/gitlab/message-sender.js]: can you redirect this ? read about different status code

File: api/gitlab/gitlab-entity.js [hunk 9]
```diff
@@ -0,0 +1,7 @@
+const GITLAB_ENTITY = {
```
Comment (dvijeshpatel) [inline api/gitlab/gitlab-entity.js]: think before giving name. design again

File: api/gitlab/name-fetcher.js [hunk 10]
```diff
@@ -0,0 +1,11 @@
+const request = require("request");
+
+module.exports = function(id) {
```
Comment (dvijeshpatel) [inline api/gitlab/name-fetcher.js]: is it used?

File: api/slack/send-message.js [hunk 11]
```diff
@@ -20,9 +20,12 @@ const SLACK_ENTITY_TO_BOT_API = {
     [SLACK_ENTITY.CHANNEL]: 'postMessageToChannel'
 }
 
-router.get('/', function(req,res) {
+router.post('/', function(req,res) {
     const { type, name, message } = req.body;
-    bot[SLACK_ENTITY_TO_BOT_API[type]](name,message);
+    const params = {
```
Comment (dvijeshpatel) [inline api/slack/send-message.js]: move outside this function give name SLACK_PARAMS

**Pull request merged**